src/scenic/core/scenarios.py

```diff
@@ -1,518 +1,595 @@
+    simulator you want to use, and pass the scene to its `simulate` method.
+
+    Attributes:
+        objects (tuple of :obj:`~scenic.core.object_types.Object`): All objects in the
+          scene. The ``ego`` object is first, if there is one.
+        egoObject (:obj:`~scenic.core.object_types.Object` or `None`): The ``ego`` object, if any.
+        params (dict): Dictionary mapping the name of each global parameter to its value.
+        workspace (:obj:`~scenic.core.workspaces.Workspace`): The :term:`workspace` for the scenario.
+
+    .. versionchanged:: 3.0
+
+        The ``egoObject`` attribute can now be `None`.
+    """
+    def __init__(self, workspace, objects, egoObject, params,
+                 temporalReqs=(),terminationConds=(), termSimulationConds=(),
+                 recordedExprs=(), recordedInitialExprs=(), recordedFinalExprs=(),
+                 monitors=(), behaviorNamespaces={}, dynamicScenario=None,
+                 sample={}, compileOptions={}):
+        self.workspace = workspace
+        self.objects = tuple(objects)
+        self.egoObject = egoObject
+        self.params = params
+        self.temporalRequirements = tuple(temporalReqs)
+        self.terminationConditions = tuple(terminationConds)
+        self.terminateSimulationConditions = tuple(termSimulationConds)
+        self.recordedExprs = tuple(recordedExprs)
+        self.recordedInitialExprs = tuple(recordedInitialExprs)
+        self.recordedFinalExprs = tuple(recordedFinalExprs)
+        self.monitors = tuple(monitors)
+        self.behaviorNamespaces = behaviorNamespaces
+        self.dynamicScenario = dynamicScenario
+        self.sample = sample
+        self.compileOptions = compileOptions
+
+    def dumpAsScenicCode(self, stream=sys.stdout):
+        """Dump Scenic code reproducing this scene to the given stream.
+
+        For non-human-readable but complete serialization of scenes see
+        `Scenario.sceneToBytes` and `Scenario.sceneFromBytes`.
+
+        .. note::
+
+            This function does not currently reproduce parts of the original Scenic
+            program defining behaviors, functions, etc. used in the scene. Also, if
+            the scene involves any user-defined types, they must provide a suitable
+            :obj:`~object.__repr__` for this function to print them properly.
+
+        Args:
+            stream (:term:`text file`): Where to print the code (default `sys.stdout`).
+        """
+        for name, value in self.params.items():
+            if str.isidentifier(name):
+                stream.write(f'param {name} = ')
+            else:
+                stream.write(f'param "{name}" = ')
+            dumpAsScenicCode(value, stream)
+            stream.write('\n')
+        stream.write('ego = ')
+        for obj in self.objects:
+            dumpAsScenicCode(obj, stream)
+            stream.write('\n')
+
+    def show(self, zoom=None, block=True):
```

Comment:
If the 3D viewer is not going to support these options we should remove them from `show`. But `zoom` at least does make sense for the 3D viewer. It's less important now that the user can zoom in and out from the GUI, but still could be useful for scenarios with widely-scattered objects where it might be hard to find the ego.